python/ray/tune/utils/file_transfer.py

```diff
@@ -0,0 +1,234 @@
+import io
```

Comment:
Add unit tests for public functions?

Reply:
`sync_dir_between_nodes` is tested in `test_sync` (as part of the `RemoteTaskSyncer`). I can add a test for `delete_on_node`

Reply:
Added